### PR TITLE
Refactor, add doc and tests, improve installation, add an option to remove some tweet's trailing URL

### PR DESCRIPTION
**Pull request by fcayre**
- follow standard coding style
- refactor to improve readabilty and reduce code duplication
- add docstrings
- make t2m pip installable ("pip install ." or "python setup.py install" on a git clone, in a virtualenv or not)
- add unit tests and a tox configuration to test on python 2.7 and 3.5
- add a "-s" option to strip the trailing URL that Twitter API adds to the end of some tweets (see https://twittercommunity.com/t/api-returns-url-to-twitters-status-update-at-the-end-of-the-text/50424)

### CODE REVIEW

**Comment by fcayre** [comment]
I also added a travis-ci configuration, which status on my branch is:
[Build Status](https://travis-ci.org/fcayre/t2m)

**Comment by Psycojoker** [comment]
Hello,

Thanks a lot for your work :) As you might have guessed, I'm sadly quite busy right now so I don't know when I'll have the time to correctly review that (and this is not a small PR), sorry for that :confused:

File: tests/data/t2m_mamot.fr_clientcred.txt [hunk 1]
```diff
@@ -0,0 +1,2 @@
+2e047002b6d247c2712f12ebb30f61283d3b2c8b41219ac26e964539f2734bf2
+35865d973db3477963b622d99f2f2c0fbde2ef6aa8840aa7082b9ab7ecc6efa9
```
Comment (Psycojoker) [inline tests/data/t2m_mamot.fr_clientcred.txt]: You might want to force push on your branch to remove this and revoke them :x

Comment (fcayre) [reply tests/data/t2m_mamot.fr_clientcred.txt]: @Psycojoker just removed them, pushed by error but revoked, thanks. Was not needed for the tests of course.

**Comment by fcayre** [comment]
No hurry, take the time you need!

**Comment by fcayre** [comment]
Hi @Psycojoker, will you any time to review my PR in the coming days/ weeks?

**Comment by Psycojoker** [comment]
Finally to the time to merge it, sorry for the delay :sweat_smile:

And thanks a lot for your contribution, this will help release t2m on pypi :)

**Comment by Psycojoker** [comment]
Btw, if you are interesting in helping maintaining this tool, it's now part of [YoloSwagTeam](http://yoloswag.team/) if this is somehting in which you would find your place :)

**Pull request merged**